database/database.go

```diff
@@ -333,10 +347,10 @@ func ensureDBSchema(ctx context.Context, db *mongo.Database, log *logrus.Logger)
 				Keys:    bson.D{{"skylink", 1}},
 				Options: options.Index().SetName("skylink").SetUnique(true),
 			},
-			{
-				Keys:    bson.D{{"failed", 1}},
-				Options: options.Index().SetName("failed"),
-			},
+			// {
```

Comment:
I disabled this index for the time being. This was added in the last deploy. This is a bit of a stretch obviously but maybe the reason this broke it is because of the index name?? If I remove this index the deadlock disappears. 

It makes no sense though because I tested this manually on `dev1` and `eu-pol-5` before merging it.
That index was added then as well and it was not an issue when I manually built the container and
recreated it during testing.